### PR TITLE
slice: basic implementation

### PR DESCRIPTION
**Pull request by ninedraft**

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package golds
+
+// Slice is a sequence of values.
+// Basically it's an utility wrapper around a plain go slice.
```
Comment (cristaloleg) [inline slice.go]: s/go/''

File: slice.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+// Repeat returns a new Slice[E] with n copies of v.
+func repeat[E any](n int, v E) slice[E] {
+	switch{
+	case n == 0:
+		return nil
+	case n < 0:
+		panic("golds.Repeat: negative Repeat count")
+	}
+	var s = make(slice[E], n)
```
Comment (cristaloleg) [inline slice.go:21]: can be in default-case

File: slice.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+func(s slice[E]) Len() int { return len(s) }
+
+// Cap returns capacity of underlying slice.
+func(s slice[E]) Cap() int { return cap(s) }
+
+// Index returns element by the index.
+// If index is negative, then returns element len-index.
+// Panics if index is out of [-len, len] range.
+func(s slice[E]) Index(i int) E {
+	if i < 0 {
```
Comment (cristaloleg) [inline slice.go:54]: what about `|i| > len(s)` ? :)

**Review by cristaloleg — COMMENTED** [review]
sgtm

File: slice.go (hunk 3 above)
Comment (ninedraft) [reply slice.go:54]: panek

File: slice.go (hunk 2 above)
Comment (ninedraft) [reply slice.go:21]: I want to provide meaningful panic message for n < 0 cases

File: slice.go (hunk 1 above)
Comment (ninedraft) [reply slice.go]: :heavy_check_mark:

**Review by cristaloleg — APPROVED** [review]

**Pull request merged**